microbench/lower_bound_benchmark.cpp

```diff
@@ -0,0 +1,56 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2021 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Andrey Abramov
+////////////////////////////////////////////////////////////////////////////////
+
+#include <benchmark/benchmark.h>
+#include <numeric>
+
+namespace {
+
+void BM_lower_bound(benchmark::State& state) {
```

Comment:
Btw interesting fact we can change asssimpotic to O(log(k)) where k is position in array
just change choose "m" strategy: 1 2 4...2^i and then find bigger value. You can use common binary search between last position and current

This algorithm more tolerant to value at the beginning and better in comparable of linear search for small founded index